python/dgl/data/qm9.py

```diff
@@ -0,0 +1,173 @@
+"""QM9 dataset for graph property prediction (regression)."""
+import os
+import numpy as np
+import scipy.sparse as sp
+import torch
+
+from .dgl_dataset import DGLDataset
+from .utils import download, _get_dgl_url
+from ..convert import graph as dgl_graph
+from ..transform import to_bidirected
+
+class QM9Dataset(DGLDataset):
+    r"""QM9 dataset for graph property prediction (regression)
+
+    This dataset consists of 13,0831 molecules with 12 regression targets.
+    Node means atom and edge means bond.
+
+    Reference: `"Directional Message Passing for Molecular Graphs" <https://arxiv.org/abs/2003.03123>`_
+    
+    Statistics:
+
+    - Number of graphs: 13,0831
+    - Number of regression targets: 12
+
+    +--------+----------------------------------+-----------------------------------------------------------------------------------+---------------------------------------------+
```

Comment:
You may want to build the doc and ensure that it looks fine. To build the doc, go to `dgl/docs/` and follow the instructions in README.